dockers/docker-sonic-restapi/cert_monitor.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python
+
+'''
+This script monitors the certificate needed by rest-api container and safely restarts
+the process when it detects a certificate roll-over
+'''
+import os, subprocess, time
+from swsscommon import swsscommon
+from sonic_daemon_base.daemon_base import Logger
+import json
+
+RESTAPI_SERVER_PROCESS = "go-server-server"
+PATH_TO_CERT_TRACKER = "/restapi/cert_tracker"
+POLLING_FREQUENCY = 3600
+REDIS_TIMEOUT_MS = 0
+logger = Logger()
+
+def get_pid(process_name):
+    '''
+    return Process ID of the given process
+    '''
+    cmd = "pgrep -f "+process_name
+    response = subprocess.Popen(cmd.split(), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    stdout, stderr = response.communicate()
+    if (response.returncode != 0):
+        logger.log_error(stderr, True)
+        return None
+    return stdout
+
+def monitor_certificates(certificates, path_to_cert_tracker):
+    '''
+    Continuously monitor for change in 'last modified' time of the certs and restart restapi if
+    change is detected
+    '''
+    while True:
+        if os.path.isfile(path_to_cert_tracker):
+            # load last read 'last modified' times
+            cert_tracker = json.load(open(path_to_cert_tracker))
+        else:
+            cert_tracker = {}
+        restart = False
+        for certificate in certificates:
+            if os.path.isfile(certificate):
+                # Get 'last modified' time of the symbolic link of the cert
+                m_time = os.lstat(certificate).st_mtime
+                if certificate in cert_tracker:
+                    if cert_tracker[certificate] != m_time:
+                        logger.log_info("monitor_certificates: Certificate has rolled. Restart required")
```

Comment:
can you put more detailed information, for example, file name, original time, new time, for better debugging purpose.